src/typings/electron-ipc.d.ts

```diff
@@ -90,7 +90,7 @@ export interface IpcRendererEvents {
   'k8s-check-state': (state: import('@/k8s-engine/k8s').State) => void;
   'k8s-current-engine': (engine: import('@/config/settings').ContainerEngine) => void;
   'k8s-current-port': (port: number) => void;
-  'k8s-restart-required': (required: Record<string, [any, any] | []>) => void;
+  'k8s-restart-required': (required: Awaited<ReturnType<import('@/k8s-engine/k8s').KubernetesBackend['requiresRestartReasons']>>) => void;
```

Comment:
This needs a comment to explain why `Awaited` is needed.

Let me try to work it out:

1. The return type of `KubernetesBackend.requiresRestartReasons` is changed from 
`Promise<Record<string, [any, any] | []>` to
`Promise<Record<string, RestartReason | undefined>>`

No reason for the `Awaited` keyword here.

2. So I assume the event type definition for `k8s-restart-required` was incorrect. I don't run into problems if I change it to `Record<string, RestartReason | undefined>` -- but I gather this is incorrect. The typescript compiler (or whatever) isn't complaining, but it's wrong, because the thenable nature of this type is lost -- correct? It doesn't affect behaviour of the system, because event-processing is already async (I'm guessing), but you're trying to provide the correct type. Or maybe when you used `<ReturnType<import('@/k8s-engine/k8s').KubernetesBackend['requiresRestartReasons']>>) => void` you were forced to use `Awaited`. I just want a bit of clarification.